src/Extensions.hs

```diff
@@ -180,31 +175,33 @@ getModuleExtentionsBySource cabalFile path source =
 ----------------------------------------------------------------------------
 
 mergeCabalAndModule
-    :: [OnOffExtension]
+    :: ParsedExtensions
     -> FilePath
-    -> Either ParseError [OnOffExtension]
+    -> Either ParseError ParsedExtensions
     -> ExtensionsResult
 mergeCabalAndModule cabalExts path moduleRes = case moduleRes of
-    Right moduleExts -> Right $ mergeExtensions $ cabalExts <> moduleExts
+    Right moduleExts -> case mergeAnyExtensions cabalExts moduleExts of
+        Just res -> Right res
+        Nothing  -> Left SafeHaskellConflict
```

Comment:
I think the `mergeAnyExtensions` function can return `Either (.., ...) Extensions` and we can add more info in this error case. What do you think?